src/picongpu/include/particles/ionization/byCollision/ThomasFermi/AlgorithmThomasFermi.hpp

```diff
@@ -58,7 +58,14 @@ namespace ionization
     {
         /** Functor implementation
          *
-         * \tparam DensityType type of number
+         * Calling this functor gives a prediction on the ion charge state
```

Comment:
I would make the `operator()` a `DINLINE` member function and add the `if (prevBoundElectrons > newBoundElectrons)` check in the operator to make this a real ionizer algorithm and not a detailed balance functor